File: 코테준비/프로그래머스/사전/폰켓몬.js
function solution(nums) {
    var answer = 0;
    var dic =  {};
    for (let num of nums) {
        if (num in dic) {
            dic[num] = dic[num] + 1;
        }
        else {
            dic[num] = 1;
        }
    }
    n = parseInt(nums.length/2);
    
    if (n > Object.keys(dic).length) {
        return Object.keys(dic).length
    }
    return n;
}
let nums = [3,1,2,3]
console.log(solution(nums))

File: 코테준비/프로그래머스/사전/폰켓몬.py
from collections import defaultdict


def solution(nums):
    dic = defaultdict(int)
    for num in nums:
        dic[num] += 1
    n = len(nums) // 2
    dic_num = len(dic)
    if dic_num < n:
        return dic_num

    return n


nums = [3, 1, 2, 3]

print(solution(nums))
